rust/agama-cli/src/main.rs

```diff
@@ -139,12 +151,24 @@ async fn build_manager<'a>() -> anyhow::Result<ManagerClient<'a>> {
 }
 
 async fn run_command(cli: Cli) -> Result<(), ServiceError> {
+    // we need to distinguish commands on those which assume that authentication JWT is already
+    // available and those which not (or don't need it)
+    let mut client = if let Commands::Auth(_) = cli.command {
+        BaseHTTPClient::default()
+    } else {
+        // this deals with authentication need inside
+        BaseHTTPClient::new()?
+    };
+
+    client.base_url = cli
+        .opts
+        .api
+        .strip_suffix("/")
```

Comment:
This returns `None` if the string has no such suffix, and then I keep wondering why my `--api` gets ignored
https://doc.rust-lang.org/nightly/core/primitive.str.html#method.strip_suffix

https://doc.rust-lang.org/nightly/core/primitive.str.html#method.trim_end_matches should work better